webviz_subsurface/plugins/_group_tree/_views/_group_tree_view/_view.py

```diff
@@ -0,0 +1,350 @@
+                                {"label": "P10 (high)", "value": StatOptions.P10.value},
+                                {
+                                    "label": "P50 (median)",
+                                    "value": StatOptions.P50.value,
+                                },
+                                {"label": "P90 (low)", "value": StatOptions.P90.value},
+                                {"label": "Maximum", "value": StatOptions.MAX.value},
+                                {"label": "Minimum", "value": StatOptions.MIN.value},
+                            ],
+                        )
+                    ],
+                ),
+                html.Div(
+                    id=self.register_component_unique_id(
+                        ViewOptions.Ids.SINGLE_REAL_OPTIONS
+                    ),
+                    children=[
+                        wcc.Dropdown(
+                            label="Realization",
+                            id=self.register_component_unique_id(
+                                ViewOptions.Ids.REALIZATION
+                            ),
+                            options=[],
+                            value=None,
+                            multi=False,
+                        )
+                    ],
+                ),
+            ],
+        )
+
+
+class ViewFilters(SettingsGroupABC):
+    class Ids(str, Enum):
+        TOUR_STEP = "tour-step"
+        PROD_INJ_OTHER = "prod-inj-other"
+
+    def __init__(self) -> None:
+        super().__init__("Filters")
+
+    def layout(self) -> html.Div:
+        return html.Div(
+            id=self.register_component_unique_id(self.Ids.TOUR_STEP),
+            children=[
+                wcc.SelectWithLabel(
+                    id=self.register_component_unique_id(self.Ids.PROD_INJ_OTHER),
+                    label="Prod/Inj/Other",
+                    options=[
+                        {"label": "Production", "value": NodeType.PROD.value},
+                        {"label": "Injection", "value": NodeType.INJ.value},
+                        {"label": "Other", "value": NodeType.OTHER.value},
+                    ],
+                    value=[
+                        NodeType.PROD.value,
+                        NodeType.INJ.value,
+                        NodeType.OTHER.value,
+                    ],
+                    multi=True,
+                    size=3,
+                )
+            ],
+        )
+
+
+class GroupTreeView(ViewABC):
+    class Ids(str, Enum):
+        VIEW_ELEMENT = "view-element"
+        CONTROLS = "controls"
+        OPTIONS = "options"
+        FILTERS = "filters"
+        GROUPTREE_COMPONENT = "grouptree-component"
+
+    def __init__(self, group_tree_data: Dict[str, EnsembleGroupTreeData]) -> None:
+        super().__init__("Group Tree")
+        self._group_tree_data = group_tree_data
+        self._group_tree_component_id = "grouptree-component"
```

Comment:
The problem here is that the view class doesn't have a `register_component_unique_id` class. AND this id is actually not used for anything, it is just because the `GroupTree` component in `webviz-subsurface-components` must have an id input.

Reply:
You're right. Then use the view element's method instead (you'd have to move the definition behind of the `add_view_element` call):
```python
self._group_tree_component_id = self.view_element(GroupTreeView.Ids.VIEW_ELEMENT).register_component_unique_id(GroupTreeView.Ids.GROUPTREE_COMPONENT)
```

Reply:
Yes, that works!